app/controllers/NCTSMessageController.scala

```diff
@@ -57,14 +57,23 @@ class NCTSMessageController @Inject()(cc: ControllerComponents, getArrival: GetA
 
       messageResponse match {
         case Some(response) =>
-          val newState = request.arrival.status.transition(response.messageReceived)
-          saveMessageService.validateXmlAndSaveMessage(xml, messageSender, response, newState) map {
-            case SubmissionSuccess         => Ok
-            case SubmissionFailureInternal => InternalServerError
-            case SubmissionFailureExternal => BadRequest
+          val newState         = request.arrival.status.transition(response.messageReceived)
+          val processingResult = saveMessageService.validateXmlAndSaveMessage(xml, messageSender, response, newState)
+          processingResult map {
+            case SubmissionSuccess => Ok
+            case SubmissionFailureInternal =>
+              val message = "Internal Submission Failure " + processingResult
```

Comment:
The `processingResult ` isn't going to give much information here aside from if it was a `SubmissionFailureInternal` vs a `SubmissionFailureExternal`, but if that's what's wanted then its fine